include/experimental/__p1673_bits/conjugated.hpp

```diff
@@ -44,135 +44,92 @@
-    return conj(val) + upd;
-  }
-
-  template<class T2>
-  bool operator== (const T2 upd) const {
-    return conj(val) == upd;
-  }
-
-  template<class T2>
-  bool operator!= (const T2 upd) const {
-    return conj(val) != upd;
-  }
-
-private:
-  const T& val;
-};
-
-template<class T1, class T2>
-auto operator* (const T1 x, const conjugated_scalar<T2> y) {
-  using std::conj;
-  return x * T2(y);
-}
-
-template<class Accessor, class T>
+template<class Accessor>
 class accessor_conjugate {
 private:
-  using size_type = typename extents<>::size_type;
-public:
-  using element_type  = typename Accessor::element_type;
-  using pointer       = typename Accessor::pointer;
-  using reference     = typename Accessor::reference;
-  using offset_policy = typename Accessor::offset_policy;
-
-  accessor_conjugate() = default;
+  Accessor accessor_;
 
-  accessor_conjugate(Accessor a) : acc(a) {}
+  using accessor_value_type =
+    std::remove_cv_t<typename Accessor::element_type>;
+  static constexpr bool accessor_value_is_arith =
+    std::is_arithmetic_v<accessor_value_type>;
 
-  reference access(pointer p, size_type i) const noexcept {
-    return reference(acc.access(p, i));
-  }
-
-  typename offset_policy::pointer offset(pointer p, size_type i) const noexcept {
-    return acc.offset(p, i);
-  }
-
-  element_type* decay(pointer p) const noexcept {
-    return acc.decay(p);
-  }
-private:
-  Accessor acc;
-};
+public:
+  using reference =
+    std::conditional_t<accessor_value_is_arith,
+      typename Accessor::reference,
+      conjugated_scalar<typename Accessor::reference, accessor_value_type>>;
 
-template<class Accessor, class T>
-class accessor_conjugate<Accessor, std::complex<T>> {
 private:
-  using size_type = typename extents<>::size_type;
+  using pre_const_element_type =
+    std::conditional_t<accessor_value_is_arith,
+      typename Accessor::element_type,
+      typename reference::value_type>;
+  
 public:
-  // FIXME If BLAS functions want to strip off accessor_conjugate for
-  // optimization, they will need a way to work with th underlying
-  // Accessor (which may not be the default one).
-
-  using element_type  = typename Accessor::element_type;
+  using element_type  = std::add_const_t<pre_const_element_type>;
   using pointer       = typename Accessor::pointer;
-  // FIXME Do we actually need to template conjugated_scalar
-  // on the Reference type as well as T ?
-  using reference     =
-    conjugated_scalar< /* typename Accessor::reference, */ std::complex<T>>;
   using offset_policy =
-    accessor_conjugate<typename Accessor::offset_policy, std::complex<T>>;
-
-  accessor_conjugate() = default;
-
-  accessor_conjugate(Accessor a) : acc(a) {}
-
-  reference access(pointer p, size_type i) const noexcept {
-    return reference(acc.access(p, i));
+    std::conditional_t<accessor_value_is_arith,
+		       typename Accessor::offset_policy,
+		       accessor_conjugate<typename Accessor::offset_policy>>;
+
```

Comment:
It would be good to keep the default constructor - without it something in my code is broken.

Reply:
@youyu3 Default constructibility isn't one of the accessor policy requirements, so I would prefer not to require it here.  Please coordinate with me offline to see if it's possible to work around.

Reply:
Update: it was possible to work around, yay!